### PR TITLE
Add com.github.lainsce.notejot

### PR DESCRIPTION
**Pull request by lainsce**
This is the file that makes possible to build a Flatpak for Notejot, my own sticky notes application made for elementary OS, but now in Flatpak format.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build com.github.lainsce.notejot

File: com.github.lainsce.notejot.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    "build-options": {
+        "env": {
+            "PKG_CONFIG_GOBJECT_INTROSPECTION_1_0_GIRDIR": "/app/share/gir-1.0",
+            "PKG_CONFIG_GOBJECT_INTROSPECTION_1_0_TYPELIBDIR": "/app/lib/girepository-1.0"
+        }
+    },
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--device=all",
```
Comment (nedrichards) [inline com.github.lainsce.notejot.json]: what do you need device=all for?

File: com.github.lainsce.notejot.json [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        "--socket=wayland",
+        "--filesystem=home",
```
Comment (nedrichards) [inline com.github.lainsce.notejot.json]: what do you need access to all of home for?

Comment (lainsce) [reply com.github.lainsce.notejot.json]: For cache purposes, due to it saving to `.local/<name_of_app>/cache.json`

File: com.github.lainsce.notejot.json (hunk 1 above)
Comment (lainsce) [reply com.github.lainsce.notejot.json]: I actually saw this in other elementary apps' flatpak json files, might be something that's needed. I couldn't complete the flatpak build and make it work without it. 🤔

File: com.github.lainsce.notejot.json (hunk 2 above)
Comment (nedrichards) [reply com.github.lainsce.notejot.json]: Right, but you don't need all of home permission to save to a single directory path, look at the dconf section further down the manifest for an example.

And even then @TingPing will be along soon to talk about keeping this stuff within the flatpak itself ;-)

Comment (lainsce) [reply com.github.lainsce.notejot.json]: I mean, I'm still a newbie at flatpakking stuff up, so any pointers help 👍 
I'm going to look that dconf section up and see how to do it better.

File: com.github.lainsce.notejot.json (hunk 1 above)
Comment (nedrichards) [reply com.github.lainsce.notejot.json]: What errors do you get? notes-up, for example doesn't have it. https://github.com/flathub/com.github.philip_scott.notes-up/blob/master/com.github.philip_scott.notes-up.json

we needed it in spice-up to enable access to remote presentation devices, it's mostly if you require specific hardware support that doesn't yet have a portal.

**Comment by nedrichards** [comment]
bot, build com.github.lainsce.notejot

File: com.github.lainsce.notejot.json (hunk 1 above)
Comment (lainsce) [reply com.github.lainsce.notejot.json]: I guess it was a one-time error, I could build it now when I removed it. I'm going to do a new push without the line.

**Comment by lainsce** [comment]
I'm still fixing the file lol there's so much stuff I missed.

**Comment by nedrichards** [comment]
No worries, it's a learning experience! I'm delighted that developers are interested in using flathub - that's one of its key reasons to exist, to help developers reach users directly, so happy to help.

**Comment by lainsce** [comment]
I guess now it's all fixed 😉

**Comment by nedrichards** [comment]
bot, build com.github.lainsce.notejot

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.github.lainsce.notejot

**Pull request closed without merging**